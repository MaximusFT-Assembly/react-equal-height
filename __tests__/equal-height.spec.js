import React, { useContext, useState } from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import { EqualHeightConsumer, EqualHeight, EqualHeightElement, EqualHeightContext } from '../';

describe('EqualHeight', () => {
    test('should render correctly', () => {
        const container = shallow(
            <EqualHeight children={null} />
        );

        expect(container).toMatchSnapshot();
    });

    test('should update state on resize event', () => {
        let resizeStatus = 'NOT INITIALIZED';

        render(
            <EqualHeight timeout={0}>
                <EqualHeightConsumer>
                    {context => {
                        resizeStatus = context.update;
                    }}
                </EqualHeightConsumer>
            </EqualHeight>
        );

        expect(resizeStatus).not.toBe('NOT INITIALIZED');
        expect(resizeStatus).toBe(true);

        act(() => {
            window.dispatchEvent(new Event('resize'));
        });

        expect(resizeStatus).toBe(false);
    });

    test('should update state on orientationchange event', () => {
        let orientationChangeStatus = 'NOT INITIALIZED';

        render(
            <EqualHeight timeout={0}>
                <EqualHeightConsumer>
                    {context => orientationChangeStatus = context.update}
                </EqualHeightConsumer>
            </EqualHeight>
        );

        expect(orientationChangeStatus).not.toBe('NOT INITIALIZED');
        expect(orientationChangeStatus).toBe(true);

        act(() => {
            window.dispatchEvent(new Event('orientationchange'));
        });

        expect(orientationChangeStatus).toBe(false);
    });

    test('should update state on forceUpdate method call', () => {
        let forceUpdateStatus = 'NOT INITIALIZED';
        let updateStatus = 'NOT INITIALIZED';

        const MockForceUpdate = () => {
            const { setForceUpdate } = useContext(EqualHeightContext);

            return (
                <p data-testid={'forceUpdateElement'} onClick={() => setForceUpdate(value => !value)} />
            );
        };

        const { getByTestId } = render(
            <EqualHeight>
                <EqualHeightConsumer>
                    {context => {
                        forceUpdateStatus = context.forceUpdate;
                        updateStatus = context.update;
                    }}
                </EqualHeightConsumer>
                <MockForceUpdate />
            </EqualHeight>
        );

        expect(forceUpdateStatus).not.toBe('NOT INITIALIZED');
        expect(updateStatus).not.toBe('NOT INITIALIZED');

        expect(forceUpdateStatus).toBe(false);
        expect(updateStatus).toBe(true);

        fireEvent.click(getByTestId('forceUpdateElement'));

        expect(forceUpdateStatus).toBe(true);
        expect(updateStatus).toBe(false);
    });
});
