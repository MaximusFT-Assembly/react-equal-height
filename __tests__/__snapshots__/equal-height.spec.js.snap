// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EqualHeight should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "animationSpeed": 0.25,
      "childrenCount": 0,
      "forceUpdate": false,
      "originalChildrenCount": 0,
      "setChildrenCount": [Function],
      "setForceUpdate": [Function],
      "setOriginalChildrenCount": [Function],
      "setTemporarySizes": [Function],
      "sizes": Array [],
      "temporarySizes": Array [],
      "update": false,
      "updateOnChange": undefined,
    }
  }
/>
`;
